tests/test_database_to_staging.py

```diff
@@ -1045,3 +1046,608 @@ def test_store_data_gene_matrix(syn, wes_seqassayids, expected_output):
+                    "YEAR_CONTACT": [
+                        2033,
+                        2030,
+                        2010,
+                        2082,
+                    ],  # difference to BIRTH_YEAR: 41, 37, 18, 89
+                    "YEAR_DEATH": [
+                        2044,
+                        2047,
+                        2012,
+                        2082,
+                    ],  # difference to BIRTH_YEAR: 52, 54, 20, 89
+                    "AGE_AT_SEQ_REPORT": [
+                        13870,
+                        17885,
+                        6935,
+                        32485,
+                    ],  # in years 38*365, 49*365, 19*365, 89*365
+                    "INT_CONTACT": [
+                        14966,
+                        13506,
+                        6571,
+                        32485,
+                    ],  # in years: 41*365+1, 37*365+1, 18*365+1, 89*365
+                    "INT_DOD": [
+                        18981,
+                        19712,
+                        7300,
+                        32485,
+                    ],  # in years: 52*365+1, 54*365+2, 20*365, 89*365
+                    "Other_col": ["a", "b", "c", "d"],
+                }
+            ),
+            "expected_df": pd.DataFrame(
+                {
+                    "PATIENT_ID": ["a", "a", "c", "d"],
+                    "SAMPLE_ID": ["a", "b", "c", "d"],
+                    "BIRTH_YEAR": [1992, 1993, 1992, 1993],
+                    "YEAR_CONTACT": [
+                        2033,
+                        2030,
+                        2010,
+                        2082,
+                    ],  # difference to BIRTH_YEAR: 41, 37, 18, 89
+                    "YEAR_DEATH": [
+                        2044,
+                        2047,
+                        2012,
+                        2082,
+                    ],  # difference to BIRTH_YEAR: 52, 54, 20, 89
+                    "AGE_AT_SEQ_REPORT": [
+                        13870,
+                        17885,
+                        6935,
+                        32485,
+                    ],  # in years 38*365, 49*365, 19*365, 89*365
+                    "INT_CONTACT": [
+                        14966,
+                        13506,
+                        6571,
+                        32485,
+                    ],  # in years: 41*365+1, 37*365+1, 18*365+1, 89*365
+                    "INT_DOD": [
+                        18981,
+                        19712,
+                        7300,
+                        32485,
+                    ],  # in years: 52*365+1, 54*365+2, 20*365, 89*365
+                    "Other_col": ["a", "b", "c", "d"],
+                }
+            ),
+        },
+        {
+            "name": "has_column_to_be_redacted",
+            "input_df": pd.DataFrame(
+                {
+                    "PATIENT_ID": ["a", "b", "c", "d", "e", "e"],  # redact b, d, e
```

Comment:
@thomasyu888 this is a test case for a patient that to be redacted has multiple samples.